Library/ViewModels/DiscoveryViewModel.swift

```diff
@@ -77,10 +91,11 @@ DiscoveryViewModelOutputs {
     self.configureSortPager = self.configurePagerDataSource
 
     let currentParams = Signal.merge(
-      self.viewWillAppearProperty.signal.take(first: 1).map { _ in initialParam() },
+      self.didChangeRecommendationsSettingProperty.signal.map { _ in initialParam() },
```

Comment:
Could we rather merge the two for the `initialParam`:

```swift
let initialParam = Signal.merge(
  self.didChangeRecommendationsSettingProperty.signal,
  self.viewWillAppearProperty.signal
)
.ignoreValues()
.map { initialParam() } // if we move initialParam() to file scope this can be point-free.

let currentParams = Signal.merge(
  initialParam,
  self.filterWithParamsProperty.signal.skipNil()
)
```

Reply:
Good idea. I refactored the code following your suggestion, but had to change the implementation slightly in order to cover all edge cases.